libs/designsystem/menu/src/menu.component.ts

```diff
@@ -318,20 +320,21 @@ export class MenuComponent implements AfterViewInit, AfterContentInit, OnDestroy
     }
   }
 
-  private setupAccesibilityForUserProvidedButton() {
-    if (this.userProvidedButton) {
-      const button = this.userProvidedButton.nativeElement;
-      if (button.id) {
-        this.triggerButtonId = button.id;
-      } else {
-        this.renderer.setAttribute(button, 'id', this.triggerButtonId);
-      }
-      if (!button.getAttribute('aria-controls')) {
-        this.renderer.setAttribute(button, 'aria-controls', this.menuId);
-      }
-      if (!button.getAttribute('aria-haspopup')) {
-        this.renderer.setAttribute(button, 'aria-haspopup', 'true');
-      }
+  private setUserProvidedButtonAriaAttributes() {
+    if (!this.userProvidedButton) {
+      return;
+    }
+    const button = this.userProvidedButton.nativeElement;
```

Comment:
Consider readability (simplify guard and add space after)
```suggestion
    if (!this.userProvidedButton) return;
    
    const button = this.userProvidedButton.nativeElement;
```